### DISCUSSION TITLE
How to implement select2 in livewire?

### DISCUSSION THREAD

**Discussion by anonymouse703** [livewire/livewire#4353] [1. Help]
I added select2 via composer but when I tried to use this in my script

```
$(document).ready(function() {
    $('.js-example-basic-single').select2();
});

```
the select2 is not working on my dropdown. Do I need to use JS hooks on this?

**Comment by joshhanley** [comment]
@anonymouse703 please search through the issues and discussions on this repo, there are a few examples of how to get select2 running with Livewire 🙂

**Reply by anonymouse703** [reply]
@joshhanley  I have another question regarding select2 should I continue to ask in this thread or create new?

**Reply by joshhanley** [reply]
If its an unrelated question, then yeah a new thread is ok, but if it's related, then keep it in here 🙂

**Reply by anonymouse703** [reply]
@joshhanley What is the factor causing another branch function select2 is not working in develop but working in Job Order branch?

I already solved select2 in modal in the Job Order branch but when I tried to merged Job Order Branch in the develop the select2 in the modal is not working. When I merged the the conflict I just accept the incoming changes since the updated is from JO branch.

This is how I implemented it in the JO branch.

The script
```
document.addEventListener("DOMContentLoaded", () => {

    window.onload = function() {
        Livewire.on('select2', () => {
            $('#customer').select2({
                dropdownParent: $("#joborderModal")
            });//
        })
    }

});
```
How I called it when the I open the modal
```
public function createJobOrder()
    {
        $this->emit('resetInputFields');
        $this->emit('openJobOrderModal');
        $this->emit('select2');
    }
```
and this is the select in the modal
 ```
<!-- Customer -->
<div class="form-group">
    <label class="col-sm-4 control-label text-right" for="customer_id">Customer</label>
    <div class="col-sm-8">
        <select name="customer_id" id="customer" class="form-control" wire:model="customer_id"  required>
            <option value='' style="font-family:Courier New;text-align:center">-- SELECT CUSTOMER NAME --</option>
            @foreach ($client_profile as $data)
                <option value="{{ $data->id }}">{{ $data->name }}
                </option>
            @endforeach
        </select>
    </div>
</div>
<!-- End Customer -->
```
I already copy and paste the changes in the JO branch into the develop branch but still not working in the develop.

**Reply by joshhanley** [reply]
@anonymouse703 I'm not sure 🤷‍♂️
